static/app/views/alerts/rules/metric/triggers/chart/index.tsx

```diff
@@ -447,32 +465,165 @@ class TriggersChart extends PureComponent<Props, State> {
     };
 
     if (isOnDemandMetricAlert) {
+      const {sampleRate} = this.state;
+      const baseProps: EventsRequestProps = {
+        api,
+        organization,
+        query,
+        queryExtras,
+        sampleRate,
+        environment: environment ? [environment] : undefined,
+        project: projects.map(({id}) => Number(id)),
+        interval: `${timeWindow}m`,
+        comparisonDelta: comparisonDelta ? comparisonDelta * 60 : undefined,
+        yAxis: aggregate,
+        includePrevious: false,
+        currentSeriesNames: [formattedAggregate || aggregate],
+        partial: false,
+        includeTimeAggregation: false,
+        includeTransformedData: false,
+        limit: 15,
+        children: noop,
```

Comment:
are the components using these props, using `children` as a renderProp? (could we just pass null instead?)

Reply:
Yeah this is decidedly not clean—I think the ideal end state here (after ACI) is that these declarative `*Request` components go away.

They're already deprecated so I don't want to invest too much in cleaning this up.